Cython/Build/pep517_backend/_backend.py

```diff
@@ -0,0 +1,142 @@
+# Note - setuptools imports are all lazy so that we can specify it as a dependency
+from contextlib import contextmanager as _contextmanager
+from functools import lru_cache
+from pathlib import Path as _Path
+
+def _load_toml_file(filename: str) -> dict:
+    # I don't really want to be dealing with finding/vendoring replacement toml
+    # libraries for Python <3.11.  Therefore use setuptools if possible.
+    # Try a number of options for maximum chance of success.
+    try:
+        # pyprojecttoml is marked as private in setuptools
+        from setuptools.config.pyprojecttoml import load_file
+    except ImportError:
+        try:
+            # setuptools 69.1.0+
+            from setuptools.compat.py310 import tomllib
+        except ImportError:
+            import tomllib  # standard library from 3.11 onwards
+        def load_file(filename):
+            with open(filename, "rb") as f:
+                return tomllib.load(f)
+    return load_file(filename)
```

Comment:
You don't need to do this. Just return `tomlli; python_version < "3.11"` from the `get_requires_for_build_*()` hooks.
This library is API-compatible with `tomllib` and is maintained by a few CPython Core Devs (PEP 680 authors/sponsors). In fact, `tomllib` grew out of `tomli` so it's a very safe bet: https://realpython.com/python311-tomllib/#read-toml-with-tomllib.

```suggestion
    try:
        import tomllib
    except ImportError:
        import tomli as tomllib  # is in standard library from 3.11 onwards

    with open(filename, "rb") as f:
        return tomllib.load(f)
```